backend/person/mutations/UpdateAgentMutation.py

```diff
@@ -0,0 +1,137 @@
+from graphene import (
+    ID,
+    String,
+    InputObjectType,
+    Boolean,
+    ResolveInfo,
+    Field,
+    List,
+    Enum,
+    NonNull,
+)
+
+from person.types.AgentDescriptionType import AgentDescriptionType
+from person.models import AgentDescription, Gender
+from core.models import SourceMention
+from graphql_app.LettercraftMutation import LettercraftMutation
+from graphql_app.types.LettercraftErrorType import LettercraftErrorType
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from django.db import transaction
+from django.db.models.fields.related_descriptors import (
+    ReverseOneToOneDescriptor,
+    ForwardOneToOneDescriptor,
+)
+
+class UpdateAgentGenderInput(InputObjectType):
+    gender = Enum.from_enum(Gender)()
+    source_mention = Enum.from_enum(SourceMention)()
+    note = String()
+
+
+class UpdateAgentLocationInput(InputObjectType):
+    location = ID()
```

Comment:
This field is distinct from the `id` field, though. Because this input maps to the `through` model, `location` is a `ForeignKey` to `SpaceDescription`. `id` would be the ID of the relation object.